src/server/game/Scripting/ScriptMgr.h

```diff
@@ -1405,6 +1395,15 @@ class GenericAreaTriggerEntityScript : public AreaTriggerEntityScript
 };
 #define RegisterAreaTriggerAI(ai_name) new GenericAreaTriggerEntityScript<ai_name>(#ai_name)
 
+template <class AI>
+class GenericConversationScript : public ConversationScript
+{
+public:
+    GenericConversationScript(char const* name) : ConversationScript(name) {}
+    ConversationAI* GetAI(Conversation* convo) const override { return new AI(convo); }
```

Comment:
at some point, we might just want to stop relying on raw pointer returning for AI scripts and favour std::unique_ptr to ensure automatic memory clean-up when it goes out of scope which would reduce and *mostly* void memory leaks:

```cpp
    std::unique_ptr<ConversationAI> GetAI(Conversation* convo) const override
    {
        return std::make_unique<AI>(convo);
    }
```

Reply:
it is used as a unique_ptr, check `_ai`

Reply:
oh, I'm late to the party 😞